src/lib/Theme.ts

```diff
@@ -990,19 +1046,72 @@ export default class Theme {
                     };
                     systemPage = (await ThemeService.createAvailabePage(pageData)).data;
                 }
+
+                // Get page settings props
                 systemPage.props =
                     (
                         Theme.extractSettingsFromFile(
                             path.join(process.cwd(), 'theme', 'templates', 'pages', fileName)
                         ) || {}
                     ).props || [];
+
+                // Check if any section tag available in file
                 systemPage.sections_meta = Theme.extractSectionsFromFile(
                     path.join(process.cwd(), 'theme', 'templates', 'pages', fileName)
                 );
+                // set page type to system page
                 systemPage.type = 'system';
                 pagesToSave.push(systemPage);
             });
 
+            // remove .vue from file name
+            const allLocalSystemPageNames = systemPagesLocally.map(name => name.replace('.vue', ''))
+            // Delete system pages that were available before sync but now deleted
+            const systemPagesToDelete = systemPagesDB.filter(x => !allLocalSystemPageNames.includes(x.value));
```

Comment:
Instead of this check if we can only filter deleted pages from current available pages of a theme.

Reply:
I didn't get this, from where we will get "current available pages of a theme"?